numba/tests/test_recarray_usecases.py

```diff
@@ -144,5 +145,18 @@ def test_usecase5(self):
         self._test_usecase2to5(usecase5, self.aligned_dtype)
 
 
+class TestRecArrayCreation(unittest.TestCase):
+    def test_creation_from_tuple_list(self):
+        a_b_type = np.dtype([('a', 'i8'), ('b', 'f8')], align=True)
+        x = np.array([(1, 2.), (3, 4.)], a_b_type)
```

Comment:
Can you also add a test for
```
        a_b_type = np.dtype([('a', 'i8'), ('b', [('c', 'f8')]), align=True)
        x = np.array([(1, (2.,)), (3, (4.,))], a_b_type)
```

Reply:
sorry, I don't know what dtype that is. There seems to be a mismatch of square and round brackets, but even after fixing that, I don't think that's a valid dtype.

Reply:
Added the missing brackets and parens, whoops